src/main/java/bio/terra/service/iam/AuthenticatedUserRequest.java

```diff
@@ -67,4 +69,28 @@ public AuthenticatedUserRequest reqId(UUID reqId) {
         return this;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof AuthenticatedUserRequest)) {
+            return false;
+        }
+        AuthenticatedUserRequest that = (AuthenticatedUserRequest) o;
+        return Objects.equals(getEmail(), that.getEmail()) &&
+            Objects.equals(getSubjectId(), that.getSubjectId()) &&
+            Objects.equals(getToken(), that.getToken()) &&
+            Objects.equals(getReqId(), that.getReqId());
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17, 37)
```

Comment:
```suggestion
        return new HashCodeBuilder()
```

I think you can use the default constructor. In either case, since hashCode returns a 32-bit integer, this brings up a more general question of how many items we expect to see in the cache due to a probability of collisions. Also, my understanding is that HashCodeBuilder doesn't have great dispersal properties (increasing the collision probability). Perhaps these aren't limitations since we don't expect too many items in the cache and we do a full equality check anyways, but maybe something to think about.

Reply:
It seemed like an okay tradeoff since this is more performant. This was run past Albano as well.